Add `resistor-color-trio`

BNAndras opened the pull request:
Related to #739

BNAndras commented:
I'm thinking it'd be useful to add tests for decimal resistor values since 3.3k ohm resistors are a thing. We recently added that on the Emacs Lisp track, and I think those tests were borrowed from Julia.

https://github.com/exercism/emacs-lisp/blob/main/exercises/practice/resistor-color-trio/resistor-color-trio-test.el

jiegillet commented on exercises/practice/resistor-color-trio/.meta/src/ResistorColorTrio.example.elm:
```diff
@@ -0,0 +1,83 @@
+label colors =
+    let
+        raw =
+            colors |> rawValue |> toFloat
```
Why do you need to use floats here? Would `//` not do the job for integer division?

We got a whole thing with floats at work, they make me a bit jumpy now :)

BNAndras replied:
Alright, I wrote out the long-form literals so we don't need floats anymore.

jiegillet reviewed (approved):
Looks great, approving already.
I left small non-blocking suggestions.

jiegillet commented:
> I'm thinking it'd be useful to add tests for decimal resistor values since 3.3k ohm resistors are a thing. We recently added that on the Emacs Lisp track, and I think those tests were borrowed from Julia.

I would rather stick to the canonical data, it really makes life easier in the long term for maintainers.

BNAndras commented on config.json:
```diff
@@ -469,6 +469,22 @@
+          "custom-types",
+          "pattern-matching"
+        ],
+        "difficulty": 2
```
Sure.

BNAndras commented:
I renamed a few things and I added some handling for single-band resistor lists just as a nice nod back to `resistor-color` and how we're building up the complexity.

jiegillet commented:
Awesome :)
